src/output-json-file.c

```diff
@@ -76,6 +76,31 @@ typedef struct JsonFileLogThread_ {
     MemBuffer *buffer;
 } JsonFileLogThread;
 
+json_t *JsonFileAddMetadata(const Packet *p, uint64_t tx_id)
+{
+    json_t *fjs = json_object();
+    if (unlikely(fjs == NULL))
+        return NULL;
+
+    FileContainer *ffc = AppLayerParserGetFiles(p->proto, p->flow->alproto, p->flow->alstate, STREAM_TOCLIENT);
```

Comment:
this STREAM_TOCLIENT should probably not be hardcoded, but instead be taken from the packet direction